src/fsharp/PrettyNaming.fs

```diff
@@ -351,36 +365,40 @@ module public FSharp.Compiler.PrettyNaming
         c = '.'
         || IsIdentifierPartCharacter c
 
+    let rec isTildaOnlyString (s: string) =
```

Comment:
Is this `rec` needed?

Reply:
Nope, it was used before the logic was extracted to `loop` function.